Subject: [PATCH] Vehicle keeps accelerating.
---
Index: src/chrono_models/vehicle/hmmwv/powertrain/HMMWV_EngineShafts.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/chrono_models/vehicle/hmmwv/powertrain/HMMWV_EngineShafts.cpp b/src/chrono_models/vehicle/hmmwv/powertrain/HMMWV_EngineShafts.cpp
--- a/src/chrono_models/vehicle/hmmwv/powertrain/HMMWV_EngineShafts.cpp	(revision 9d9daf30cf89cb9624e7c882e83a8286062cc92a)
+++ b/src/chrono_models/vehicle/hmmwv/powertrain/HMMWV_EngineShafts.cpp	(date 1704230899481)
@@ -31,37 +31,16 @@
 void HMMWV_EngineShafts::SetEngineTorqueMap(std::shared_ptr<ChFunction_Recorder>& map) {
     double rpm_to_radsec = CH_C_2PI / 60.;
 
-    map->AddPoint(-100 * rpm_to_radsec, 300);  // to start engine
-    map->AddPoint(800 * rpm_to_radsec, 382);
-    map->AddPoint(900 * rpm_to_radsec, 490);
-    map->AddPoint(1000 * rpm_to_radsec, 579);
-    map->AddPoint(1100 * rpm_to_radsec, 650);
-    map->AddPoint(1200 * rpm_to_radsec, 706);
-    map->AddPoint(1300 * rpm_to_radsec, 746);
-    map->AddPoint(1400 * rpm_to_radsec, 774);
-    map->AddPoint(1500 * rpm_to_radsec, 789);
-    map->AddPoint(1600 * rpm_to_radsec, 793);
-    map->AddPoint(1700 * rpm_to_radsec, 788);
-    map->AddPoint(1800 * rpm_to_radsec, 774);
-    map->AddPoint(1900 * rpm_to_radsec, 754);
-    map->AddPoint(2000 * rpm_to_radsec, 728);
-    map->AddPoint(2100 * rpm_to_radsec, 697);
-    map->AddPoint(2200 * rpm_to_radsec, 664);
-    map->AddPoint(2300 * rpm_to_radsec, 628);
-    map->AddPoint(2400 * rpm_to_radsec, 593);
-    map->AddPoint(2500 * rpm_to_radsec, 558);
-    map->AddPoint(2700 * rpm_to_radsec, -400);  // fading out of engine torque
+    map->AddPoint(0 * rpm_to_radsec, 20);  // to start engine
+    map->AddPoint(500 * rpm_to_radsec, 400);
+    map->AddPoint(5000 * rpm_to_radsec, 400);
+    map->AddPoint(9000 * rpm_to_radsec, 0);
 }
 
 void HMMWV_EngineShafts::SetEngineLossesMap(std::shared_ptr<ChFunction_Recorder>& map) {
     double rpm_to_radsec = CH_C_2PI / 60.;
 
-    map->AddPoint(-50 * rpm_to_radsec, 30);  // it should never work in negative direction, anyway..
     map->AddPoint(0 * rpm_to_radsec, 0);
-    map->AddPoint(50 * rpm_to_radsec, -30);
-    map->AddPoint(1000 * rpm_to_radsec, -50);
-    map->AddPoint(2000 * rpm_to_radsec, -70);
-    map->AddPoint(3000 * rpm_to_radsec, -90);
 }
 
 }  // end namespace hmmwv
Index: src/demos/vehicle/wheeled_models/demo_VEH_HMMWV.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/demos/vehicle/wheeled_models/demo_VEH_HMMWV.cpp b/src/demos/vehicle/wheeled_models/demo_VEH_HMMWV.cpp
--- a/src/demos/vehicle/wheeled_models/demo_VEH_HMMWV.cpp	(revision 9d9daf30cf89cb9624e7c882e83a8286062cc92a)
+++ b/src/demos/vehicle/wheeled_models/demo_VEH_HMMWV.cpp	(date 1704230848221)
@@ -103,8 +103,8 @@
 // Rigid terrain
 RigidTerrain::PatchType terrain_model = RigidTerrain::PatchType::BOX;
 double terrainHeight = 0;      // terrain height (FLAT terrain only)
-double terrainLength = 200.0;  // size in X direction
-double terrainWidth = 200.0;   // size in Y direction
+double terrainLength = 20000.0;  // size in X direction
+double terrainWidth = 20000.0;   // size in Y direction
 
 // Point on chassis tracked by the camera
 ChVector<> trackPoint(0.0, 0.0, 1.75);
@@ -369,7 +369,7 @@
     ////    vis->EnableContactDrawing(ContactsDrawMode::CONTACT_FORCES);
     ////}
 
-    hmmwv.GetVehicle().EnableRealtime(true);
+    hmmwv.GetVehicle().EnableRealtime(false);
 
     while (vis->Run()) {
         double time = hmmwv.GetSystem()->GetChTime();
@@ -414,6 +414,14 @@
                       << std::endl;
         }
 
+        if (time > 0.55) { // lower this time to 0.35 to get the pendulum effect, car goes forward and then backward
+            driver->SetBraking(0.0);
+            driver->SetThrottle(0.0);
+        }
+        else {
+            driver->SetThrottle(1.0);
+        }
+
         // Driver inputs
         DriverInputs driver_inputs = driver->GetInputs();
 
